pkg/kubectl/cmd/util/printing.go

```diff
@@ -81,7 +83,10 @@ func OutputVersion(cmd *cobra.Command, defaultVersion string) string {
 // Requires that printer flags have been added to cmd (see AddPrinterFlags).
 func PrinterForCommand(cmd *cobra.Command) (kubectl.ResourcePrinter, bool, error) {
 	outputFormat := GetFlagString(cmd, "output")
-	templateFile := GetFlagString(cmd, "template")
+	templateFile := GetFlagString(cmd, "golang-template")
```

Comment:
Alternatives:
1. `-o template` and different flags for each template format: `--golang-template=...`, `--jsonpath-template=...`, etc.
2. `-o golang-template`, `-o jsonpath-template`, etc. with one template flag ``--template=...`.
3. `-o golang-template=...`, `-o jsonpath-template=...`, etc.

I think @thockin is in favor of (3). It does seem simpler.